src/AspNetCoreModuleV2/CommonLib/environmentvariablehelpers.h

```diff
@@ -98,9 +98,10 @@ class ENVIRONMENT_VAR_HELPERS
             dwResult = GetEnvironmentVariable(struNameBuffer.QueryStr(),
                 struValueBuffer.QueryStr(),
                 struValueBuffer.QuerySizeCCH());
-            if (struValueBuffer.IsEmpty())
+
+            if (dwResult <= 0)
```

Comment:
How was this case being hit? Were environment variables growing/changing between each server start?

Reply:
Yep we kept appending ASPNETCORE_HOSTING_STARTUP